WordPress/Classes/ViewRelated/System/Coordinators/MySitesCoordinator.swift

```diff
@@ -131,9 +131,15 @@ class MySitesCoordinator: NSObject {
     func showBlogDetails(for blog: Blog, then subsection: BlogDetailsSubsection) {
         showBlogDetails(for: blog)
 
-        if let blogDetailsViewController = navigationController.topViewController as? BlogDetailsViewController {
-            blogDetailsViewController.showDetailView(for: subsection)
+        let blogDetailsViewController: BlogDetailsViewController?
+
+        if Feature.enabled(.newNavBarAppearance) {
+            blogDetailsViewController = navigationController.topViewController?.children.first as? BlogDetailsViewController
```

Comment:
I think it might be good if we add an equivalent `showBlogDetails(for blog: Blog, then subsection: BlogDetailsSubsection)` method to `MySiteViewController`, and then we can do something like 

```
if let mySiteViewController = navigationController.topViewController as? MySiteViewController {
    mySiteViewController.showBlogDetails(for: blog, then: subsection)
}
```

This method can then tell its blog details view controller to show the correct detail view. This would mean that the MySitesCoordinator isn't relying on the MySiteViewController being structured in a particular way, with blog details as its first child. What do you think?

Reply:
Fixed in 289d017!